### PR TITLE
Override print methods

### PR DESCRIPTION
**Pull request by varunagrawal**
- Override print methods.
- Update wrapper.

### CODE REVIEW

File: gtsam/discrete/DiscreteFactor.h [hunk 1]
```diff
@@ -74,13 +74,14 @@ class GTSAM_EXPORT DiscreteFactor: public Factor {
...
-  // equals
+  /// equals
   virtual bool equals(const DiscreteFactor& lf, double tol = 1e-9) const = 0;
 
-  // print
-  virtual void print(const std::string& s = "DiscreteFactor\n",
-      const KeyFormatter& formatter = DefaultKeyFormatter) const {
-    Factor::print(s, formatter);
+  /// print
+  virtual void print(
```
Comment (dellaert) [inline gtsam/discrete/DiscreteFactor.h:81]: no virtual if override

File: gtsam/linear/GaussianBayesNet.h [hunk 2]
```diff
@@ -177,6 +177,13 @@ namespace gtsam {
      */
     VectorValues backSubstituteTranspose(const VectorValues& gx) const;
 
+    /// print graph
+    virtual void print(
```
Comment (dellaert) [inline gtsam/linear/GaussianBayesNet.h:181]: no virtual. I recommend checking the warnings in a clang CI run to be sure I did not miss anything.

File: gtsam/linear/GaussianFactor.h [hunk 3]
```diff
@@ -54,8 +54,11 @@ namespace gtsam {
     virtual ~GaussianFactor() {}
 
     // Implementing Testable interface
-    virtual void print(const std::string& s = "",
-        const KeyFormatter& formatter = DefaultKeyFormatter) const = 0;
+
+    /// print
+    virtual void print(
```
Comment (dellaert) [inline gtsam/linear/GaussianFactor.h:59]: no virtual

File: gtsam/nonlinear/NonlinearFactor.h [hunk 4]
```diff
@@ -70,8 +70,9 @@ class GTSAM_EXPORT NonlinearFactor: public Factor {
   /// @{
 
   /** print */
-  virtual void print(const std::string& s = "",
-    const KeyFormatter& keyFormatter = DefaultKeyFormatter) const;
+  virtual void print(
```
Comment (dellaert) [inline gtsam/nonlinear/NonlinearFactor.h:73]: no virtual

File: gtsam/symbolic/SymbolicBayesNet.h [hunk 5]
```diff
@@ -63,6 +63,13 @@ namespace gtsam {
     /** Check equality */
     GTSAM_EXPORT bool equals(const This& bn, double tol = 1e-9) const;
 
+    /// print
+    GTSAM_EXPORT void print(
```
Comment (dellaert) [inline gtsam/symbolic/SymbolicBayesNet.h:67]: Why is this the only one with GTSAM_EXPORT? Should we have that everywhere?

**Review by dellaert — APPROVED** [review]
Nice maintenance PR! Don't merge until you addressed comments and checked CI for warnings?

File: gtsam/linear/GaussianBayesNet.h (hunk 2 above)
Comment (varunagrawal) [reply gtsam/linear/GaussianBayesNet.h:181]: Great idea. And yes I'll get rid of the virtuals where there are overrides.

**Comment by varunagrawal** [comment]
I had to add a bunch of virtual destructors to satisfy the compiler's warnings, but all the warnings with Clang have been addressed. :tada:

**Pull request merged**